assets/js/base/components/formatted-price/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import NumberFormat from 'react-number-format';
+
+/**
+ * Internal dependencies
+ */
+import { currencyToNumberFormat } from './utils';
+
+/**
+ * Formatted price component.
+ *
+ * Takes a price and returns a formatted price using the NumberFormat component.
+ *
+ * @param {Object} props Component props.
+ */
+const FormattedPrice = ( { value, currency, onValueChange, ...props } ) => {
```

Comment:
Agree with @nerrad that the naming here could be more open-ended, but `FormattedMoney` may also be confusing.  I think `FormattedMonetaryValue` would be accurate, but that's a mouthful.

On another note, would it be possible to pull the store's existing value for `currency` in case a value isn't passed?  Thinking this may be less error-prone for future consumers of this component.

Reply:
> On another note, would it be possible to pull the store's existing value for currency in case a value isn't passed?

This happens if no currency is provided.

Reply:
`MonetaryAmount` is a term I've seen used in places such as schema.org. Running with that.